### PR TITLE
#9 bet placing

### PR DESCRIPTION
**Pull request by MarcAndreiFurtos**

### CODE REVIEW

**Review by DrBaxR — CHANGES_REQUESTED** [review]
Also fix the ESLint warnings

File: src/app/entity/Bet.ts [hunk 1]
```diff
@@ -1,9 +1,11 @@
 import { UserProfile } from './UserProfile';
+import { CompleteBetType } from './complete-bet-type.model';
 
 export interface Bet {
   betId?: number;
-  user?: UserProfile;
+  user?: UserProfile | null;
   amount: number;
-  odds: number;
+  odds?: number;
```
Comment (DanceaRadu) [inline src/app/entity/Bet.ts:8]: Why do you store the odds in the Bet ? Shouldn't those be stored in the CompleteBetType ?

File: src/app/service/event.service.ts [hunk 2]
```diff
@@ -3,58 +3,56 @@ import { HttpClient, HttpParams } from '@angular/common/http';
...
 
   private loadingSubject = new BehaviorSubject<boolean>(false);
   public loading$: Observable<boolean> = this.loadingSubject.asObservable();
 
   constructor(private http: HttpClient) {}
 
   addEvent(eventRequest: EventRequest): Observable<any> {
-    return this.http.post(this.addUrl, eventRequest);
+    return this.http.post('http://localhost:8080/api/v1/events/add', eventRequest);
+  }
```
Comment (DanceaRadu) [inline src/app/service/event.service.ts:21]: Why doesn't this use the apiUrl variable ? Why is the url hardcoded ?

File: src/app/service/event.service.ts [hunk 3]
```diff
@@ -3,58 +3,56 @@ import { HttpClient, HttpParams } from '@angular/common/http';
...
-    return this.http.delete(deleteUrl);
+    return this.http.delete(`${this.apiUrl}/delete/${eventId}`);
   }
 
   getUserProfiles(): Observable<UserProfile[]> {
-    return this.http.get<UserProfile[]>(
-      'http://localhost:8080/api/user-profiles',
-    );
+    return this.http.get<UserProfile[]>('http://localhost:8080/api/v1/user-profile');
   }
```
Comment (DanceaRadu) [inline src/app/service/event.service.ts:45]: Why is this method needed ? And wouldn't it be better placed in the user profile service ?

File: src/app/service/event.service.ts (hunk 2 above)
Comment (MarcAndreiFurtos) [reply src/app/service/event.service.ts:21]: it was a requested change

**Pull request closed without merging**